### PR TITLE
Update LIP0027

### PR DESCRIPTION
**Pull request by ricott1**

### CODE REVIEW

File: proposals/lip-0027.md [hunk 1]
```diff
@@ -104,6 +113,8 @@ This is not the case for valid binary messages properly generated by the Lisk pr
 
 A possible way to assess the validity of a binary message is to check that `encode(decode( binaryMsg )) == binaryMsg`. If this is not the case, the original binary message was not properly generated and should be rejected.
 
+**Superseded in [LIP 0064]([)https://github.com/LiskHQ/lips/blob/main/proposals/lip-0064.md)**: LIP 0064 requires that every property is _required_. Therefore, valid binary messages cannot have missing values, except for properties of type array where the missing value corresponds to the empty array, and valid messages cannot have missing properties. This implies `encode(decode( binaryMsg )) == binaryMsg` for every valid binary message `binaryMsg` and `decode(encode( msg )) == msg` for every valid message `msg`. Hence, no special care must be taken anymore.
```
Comment (ricott1) [inline proposals/lip-0027.md:116]:
```suggestion
**Superseded in [LIP 0064](https://github.com/LiskHQ/lips/blob/main/proposals/lip-0064.md)**: LIP 0064 requires that every property is _required_. Therefore, valid binary messages cannot have missing values, except for properties of type array where the missing value corresponds to the empty array, and valid messages cannot have missing properties. This implies `encode(decode( binaryMsg )) == binaryMsg` for every valid binary message `binaryMsg` and `decode(encode( msg )) == msg` for every valid message `msg`. Hence, no special care must be taken anymore.
```

File: proposals/lip-0027.md [hunk 2]
```diff
@@ -175,6 +186,8 @@ A Lisk JSON schema is a [JSON schema](https://json-schema.org/) according to [dr
 4. We restrict properties of type _array_ to [list validation](https://json-schema.org/understanding-json-schema/reference/array.html#list-validation) where each item matches the same schema. A property of type _array_ must have the <code>items</code> keyword and its value is a schema specifying the repeated schema. The value of items must include exactly one of the keywords <code>dataType</code> or <code>type</code>. This keyword must have a value from the [encoding table](#encoding-summary-table), but it is not allowed to be "array".
 5. Keywords other than `type`, `dataType`, `properties`, `items` and `fieldNumber` will be ignored by the serialization process.
 
+**Superseded in [LIP 0064]([)https://github.com/LiskHQ/lips/blob/main/proposals/lip-0064.md)**: LIP 0064 requires additionally that every schema of type _object_ must use the `required` keyword, and every property of the object must be contained its value.
```
Comment (ricott1) [inline proposals/lip-0027.md:189]:
```suggestion
**Superseded in [LIP 0064](https://github.com/LiskHQ/lips/blob/main/proposals/lip-0064.md)**: LIP 0064 requires additionally that every schema of type _object_ must use the `required` keyword, and every property of the object must be contained its value.
```

File: proposals/lip-0027.md [hunk 3]
```diff
@@ -483,25 +498,28 @@ A binary message is valid, with respect to the corresponding data and schema, if
...
+When a binary message is parsed, the binary message is decoded according to the [proto2](https://developers.google.com/protocol-buffers/docs/encoding) specifications. In particular, the following rules will be applied:
 
-Whenever a binary message does not contain a specific field, the corresponding field in the parsed object is set to its default value. The default value is type-specific:
-1. For strings, the default value is the empty string.
-2. For bytes, the default value is the empty bytes.
-3. For booleans, the default value is false.
-4. For numeric types, the default value is zero.
-5. For arrays, the default value is the empty array.
-6. For objects, the default value is an object in which each of the properties is set to its default value.
+1. Missing values: Whenever a binary message does not contain a specific field, the corresponding field in the parsed object is set to its default value. The default value is type-specific:
```
Comment (janhack) [inline proposals/lip-0027.md:515]: I think here we want to have a nested list. The markdown output is just a regular list, however. Maybe the indentation needs to be changed.

**Review by janhack — CHANGES_REQUESTED** [review]
There is one minor issue with a nested list.

Overall, I think we should avoid cosmetic formatting changes of past LIPs and only focus on the necessary changes (i.e., reference to LIP 64 in this case).

**Review by janhack — APPROVED** [review]

**Review by gkoumout — APPROVED** [review]
Looks good to me!

**Review by karmacoma — APPROVED** [review]

**Pull request merged**